openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/Endpoint.java

```diff
@@ -96,4 +100,36 @@ public boolean isDeprecated() {
 	public void setDeprecated(final boolean deprecated) {
 		this.deprecated = deprecated;
 	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) return true;
+		if (o == null || getClass() != o.getClass()) return false;
+
+		Endpoint endpoint = (Endpoint) o;
+
+		if (!Objects.equals(path, endpoint.path)) return false;
+		if (type != endpoint.type) return false;
+		if (!Objects.equals(name, endpoint.name)) return false;
+		return Objects.equals(computedName, endpoint.computedName);
+	}
+
+	@Override
+	public int hashCode() {
+		int result = path != null ? path.hashCode() : 0;
+		result = 31 * result + (type != null ? type.hashCode() : 0);
+		result = 31 * result + (name != null ? name.hashCode() : 0);
+		result = 31 * result + (computedName != null ? computedName.hashCode() : 0);
+		return result;
+	}
+
+	@Override
+	public int compareTo(Endpoint o) {
+		return Comparator
+				.comparing((Endpoint endpoint) -> endpoint.path, nullsLast(String::compareTo))
```

Comment:
Looks weird. Why the path, the type or the name could be null? Is it hiding a wrong behaviour? Like sorting before collecting these data?

Reply:
Nothing says that `path` cannot be null so better be safe than sorry. Also you do not use constructor binding so there might be a use case when you want to sort object, but it does not have path yet initialized, i.e.:

```java
var endpoint = new Enpoint();
someSortedSet.add(endpoint);
endpoint.setPath("foo");
```